tests/test_transformations.py

```diff
@@ -7,262 +7,437 @@
+            },
+            np.array([[0.0, 0.0]] * n_coords),
+            id='nparray_of_zero_coords_2d'
+        )
+    ]
+)
+def test_pix2deg_returns(kwargs, expected_value):
+    actual_value = pix2deg(**kwargs)
+    assert (actual_value == expected_value).all()
+
+
+@pytest.mark.parametrize(
+    'kwargs, expected_error',
+    [
+        pytest.param(
+            {
+                'arr': [[0] * 10],
+                'sampling_rate': 0
+            },
+            ValueError,
+            id='sampling_rate_zero_raises_value_error'
+        ),
+        pytest.param(
+            {
+                'arr': [[0] * 10],
+                'sampling_rate': -1
+            },
+            ValueError,
+            id='sampling_rate_less_zero_raises_value_error'
+        ),
+        pytest.param(
+            {
+                'arr': [0] * 5,
+                'method': 'smooth'
+            },
+            ValueError,
+            id='list_length_below_six_method_smooth_raises_value_error'
+        ),
+        pytest.param(
+            {
+                'arr': [0] * 2,
+                'method': 'neighbors'
+            },
+            ValueError,
+            id='list_length_below_three_method_neighbors_raises_value_error'
+        ),
+        pytest.param(
+            {
+                'arr': [0],
+                'method': 'preceding'
+            },
+            ValueError,
+            id='list_length_below_two_method_preceding_raises_value_error'
+        ),
+        pytest.param(
+            {
+                'arr': [0] * 10,
+                'method': 'invalid'
+            },
+            ValueError,
+            id='invalid_method_raises_value_error'
+        )
+    ]
+)
+def test_pos2vel_raises_error(kwargs, expected_error):
+    with pytest.raises(expected_error):
+        pos2vel(**kwargs)
+
+
+@pytest.mark.parametrize(
+    'method',
+    [
+        pytest.param('smooth', id='method_smooth'),
+        pytest.param('neighbors', id='method_neighbors'),
+        pytest.param('preceding', id='method_preceding')
+    ]
+)
+@pytest.mark.parametrize(
+    'kwargs, padding, expected_value',
+    [
+        pytest.param(
+            {
+                'arr': np.repeat(0, n_coords),
+                'sampling_rate': 1
+            },
+            (0, n_coords),
+            np.zeros(n_coords),
+            id='constant_input_returns_zero_velocity'
+        ),
+        pytest.param(
+            {
+                'arr': np.linspace(0, n_coords - 1, n_coords),
+                'sampling_rate': 1
+            },
+            (1, -1),
+            np.ones(n_coords),
+            id='linear_input_returns_constant_velocity'
+        )
+    ]
+)
+def test_pos2vel_returns(method, kwargs, padding, expected_value):
+    actual_value = pos2vel(**kwargs, method=method)
+    assert (actual_value[padding[0]:padding[1]] == expected_value[padding[0]:padding[1]]).all()
 
 
 def test_pos2vel_stepped_input_returns_alternating_velocity_method_preceding():
```

Comment:
one last thing:
can you parametrize these three functions too?

just call it `test_pos2vel_stepped_input_returns`

leave the creation of `N` and `x` in the function and have params like this:

```
@pytest.mark.parametrize(
    'method, expected',
    [
        pytest.param(
            'preceding', np.array([2.0, 0.0] * (N // 2)),
            id='method_preceding_alternating_velocity',
        ),
```

Reply:
@jakobchwastek @dkrako is this resolved and mergeable?

Reply:
Although this is not resolved yet, I decided to merge this PR.
I will create another issue for this remaining task.